sympy/parsing/latex/LaTeX.g4

```diff
@@ -245,8 +245,8 @@ ket: (L_BAR | BAR) expr R_ANGLE;
 mathit: CMD_MATHIT L_BRACE mathit_text R_BRACE;
 mathit_text: LETTER*;
 
-frac:
-	CMD_FRAC L_BRACE upper = expr R_BRACE L_BRACE lower = expr R_BRACE;
+frac: CMD_FRAC L_BRACE upper = expr R_BRACE L_BRACE lower = expr R_BRACE  
+	| CMD_FRAC NUMBER ;
```

Comment:
The purpose of this PR makes sense.
I'm not sure it's the best way to define syntax. `CMD_FRAC NUMBER` does more than a fraction should do, and that's why you have to deal with `\frac1234` differently in `convert_frac`.
I think it should be something like `\frac upper lower`, where both `upper` and `lower` are either a single digit or a `{expr}`.
That way you immediately fix `\frac1{2}`.

Reply:
Thanks for the quick response!

I totally agree on principal, but is it even possible to do this in ANTLR?

The problem here is that the lexer and the parser really have to talk to each other, because `12` should almost always be lexed as a single NUMBER token, unless it immediately follows a `\frac`, in which case it would be better for it to be lexed as two separate DIGIT tokens. I guess you would need a predicate that can refer to the previous token. 

Does ANTLR support this sort of thing? I am new to ANTLR myself.

Reply:
I'm giving a try myself and it seems harder than I supposed.
I'll update this PR if I succeed.
Actually I don't know much about ANTLR either, just try it intuitively using what I remember from the dragon book.

Reply:
This stuff mentioned here about 'predicates' seems relevant: https://stackoverflow.com/questions/62469159/antlr-matlab-grammar-lexing-conflict

This seems to allow some sort of custom code execution during lexing, which is what we would need.

Reply:
If we make `DIGIT` terminal and `NUMBER` non-terminal, then we can use `DIGIT` in `frac`.

Reply:
Very clever! I love it.